src/test_utils/hive/mod.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::HashMap;
+
+use ef_testing::evm_sequencer::account::{AccountType, KakarotAccount};
+use ethers::types::U256 as EthersU256;
+use eyre::eyre;
+use katana_primitives::{
+    contract::ContractAddress,
+    genesis::json::{GenesisContractJson, GenesisJson},
+};
+use reth_primitives::{Address, Bytes, B256, U256, U64};
+use serde::{Deserialize, Serialize};
+use starknet::core::utils::get_storage_var_address;
+use starknet_api::core::ClassHash;
+use starknet_crypto::FieldElement;
+
+use super::katana::genesis::{KatanaGenesisBuilder, Loaded};
+
+/// Types from https://github.com/ethereum/go-ethereum/blob/master/core/genesis.go#L49C1-L58
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct HiveGenesisConfig {
+    pub config: Config,
+    pub coinbase: Address,
+    pub difficulty: U64,
+    pub extra_data: Bytes,
+    pub gas_limit: U64,
+    pub nonce: U64,
+    pub timestamp: U64,
+    pub alloc: HashMap<Address, AccountInfo>,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Config {
+    pub chain_id: i128,
+    pub homestead_block: i128,
+    pub eip150_block: i128,
+    pub eip150_hash: B256,
+    pub eip155_block: i128,
+    pub eip158_block: i128,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct AccountInfo {
+    pub balance: U256,
+    pub code: Option<Bytes>,
+    pub storage: Option<HashMap<U256, U256>>,
+}
+
+impl HiveGenesisConfig {
+    /// Convert the [HiveGenesisConfig] into a [GenesisJson] using an [KatanaGenesisBuilder]<[Loaded]>. The [Loaded]
+    /// marker type indicates that the Kakarot contract classes need to have been loaded into the builder.
```

Comment:
This is a great comment! Nicely adapted

I do think the Loaded marker type name is a bit misleading then, but it's hard to name those:)!